frontend/src/Onboarding/TransferCoursesScreen.tsx

```diff
@@ -0,0 +1,363 @@
+          icon={<CheckBoxOutlineBlankIcon style={{ fontSize: 20 }} />}
+          checkedIcon={
+            <CheckBoxIcon style={{ fontSize: 20, color: "#EB5757" }} />
+          }
+          onChange={e => courses.forEach(course => this.onChecked(e, course))}
+        />
+        <CourseText>{allCourse.join(" and ")}</CourseText>
+      </CourseWrapper>
+    );
+  }
+
+  /**
+   * Renders course requirements in the list that are in the completed requirements list
+   * @param reqs - requirements to be rendered
+   */
+  parseCourseRequirements(reqs: IRequiredCourse[][]) {
+    return reqs
+      .filter(req =>
+        this.props.completedRequirements.some(
+          listReq =>
+            listReq.classId === req[0].classId &&
+            listReq.subject === req[0].subject
+        )
+      )
+      .map((r: IRequiredCourse[]) => this.renderCourse(r));
+  }
+
+  /**
+   * renders an entire requirement section if it has  classes specified
+   * with the title of the section
+   * Only classes that have already been selected as a completed course will be rendered
+   * @param requirementGroup - the requirement group to be rendered
+   */
+  renderSection(requirementGroup: string) {
+    const reqs = this.props.major.requirementGroupMap[requirementGroup];
+    if (!reqs || reqs.type === "RANGE") {
+      return null;
+    }
+    const renderedReqs = this.parseCourseRequirements(
+      flatten(reqs.requirements)
+    );
+    const returnDiv = (
+      <div key={requirementGroup}>
+        <TitleText>{requirementGroup}</TitleText>
+        {renderedReqs}
+      </div>
+    );
+    return renderedReqs.length > 0 ? returnDiv : null;
+  }
+
+  // renders the "Other Course" section with a button to display the add coursese modal and
+  // displays all already added courses under the button.
+  renderOtherCourseSection() {
+    return (
+      <div key="other courses">
+        <TitleText>Other Courses</TitleText>
+        <AddBlock onClick={this.showModal.bind(this)} />
+        {this.state.otherCourses.map(this.renderCourse.bind(this))}
+      </div>
+    );
+  }
+
+  render() {
+    let renderedMajorReqs = this.props.major.requirementGroups
+      .map(r => this.renderSection(r))
+      .filter(r => r !== null);
+    let split = Math.ceil(renderedMajorReqs.length / 2);
+    return (
+      <GenericOnboardingTemplate screen={2}>
+        <MainTitleText>
+          Select any courses you took as transfer credit:
+        </MainTitleText>
+        <Paper
+          elevation={0}
+          style={{
+            minWidth: 800,
+            maxWidth: 800,
+            minHeight: 300,
+            maxHeight: 300,
+            overflow: "-moz-scrollbars-vertical",
+            overflowY: "scroll",
+          }}
+          component={ScrollWrapper}
+        >
+          <Grid container justify="space-evenly">
+            <Grid key={0} item>
+              <Paper elevation={0} style={{ minWidth: 350, maxWidth: 400 }}>
```

Comment:
To add on to Andrew's comment, it seems that a lot of this is duplicated in the CompletedCoursesScreen as well. I think using a shared component would suit us well.